src/spec/map.ts

```diff
@@ -76,18 +88,19 @@ export abstract class SpecMapper<
     };
   }
 
-  protected mapFacet(spec: GenericFacetSpec<UI, LI>, params: P): GenericFacetSpec<UO, GenericLayerSpec<UO>> {
+  protected mapFacet(spec: GenericFacetSpec<UI, LI, Field>, params: P): GenericFacetSpec<UO, GenericLayerSpec<UO>, FO> {
     return {
-      ...spec,
+      // as any is required here since TS cannot infer that FO may only be FieldName or Field, but not RepeatRef
+      ...(spec as any),
       // TODO: remove "any" once we support all facet listed in https://github.com/vega/vega-lite/issues/2760
       spec: this.map(spec.spec, params) as any
     };
   }
 
-  protected mapRepeat(spec: GenericRepeatSpec<UI, LI>, params: P): GenericRepeatSpec<UO, GenericLayerSpec<UO>> {
+  protected mapRepeat(spec: RepeatSpec, params: P): GenericSpec<UO, any, RO, FO> {
     return {
       ...spec,
-      spec: this.map(spec.spec, params)
+      spec: this.map(spec.spec as any, params)
```

Comment:
comment why `as any` here (even if it's the same as in `mapFacet` -- they are two different functions).